### DISCUSSION TITLE
Theme Settings?

### DISCUSSION THREAD

**Discussion by markdav-is** [oqtane/oqtane.framework#3851] [Q&A] [answered]
I thought I heard about theme settings, but when I launch the admin tools and edit a theme, I see no settings tab.  Is there a UI for editing theme settings in a generic way or is that up to the theme developer?

**Comment by thabaum** [comment]
@markdav-is this should be up to the developer.  There are no theme settings for the default themes since they just serve a generic purpose as a starting point.  We should be able to add some extra theme settings to the Theme Management edit component.  But if we can currently and how is a great question.

I have been to review this as well so this would be good for Oqtane.Docs documentation to help shine light when heading this direction in the dark for us once we get the information to explain and demonstrate how to create these for a theme. I would imagine it would be adding an additional tab or section in theme settings.  

Question is how to inject this new tab or section in there? Is there currently a way for developers to hook onto this Theme Management component?

![image](https://github.com/oqtane/oqtane.framework/assets/13577556/8cb98ed7-7f95-476e-b8ca-f66e5ad80e51)

I will explore this soon myself to provide better feedback.

**Comment by sbwalker** [comment, marked as answer]
@markdav-is I believe you are referring to the custom Theme Settings which are displayed in Add/Edit Page and Container Settings displayed in Module Settings:

![image](https://github.com/oqtane/oqtane.framework/assets/4840590/b9acab0a-a2e5-47ae-9f09-610fab08d78b)

and

![image](https://github.com/oqtane/oqtane.framework/assets/4840590/5538be6e-5295-4cce-b255-0fcbf4c3a9f5)

These options are available for the default Oqtane Theme because it implements custom Theme Settings And Container Settings which allow a user to customize the behavior at runtime. 

If you look at the Oqtane Theme in the core framework project (Oqtane.Client\Themes\OqtaneTheme) you will notice that the ThemeInfo.cs contains a couple of type references:

```
       public Theme Theme => new Theme
       {
           Name = "Oqtane Theme",
           Version = "1.0.0",
           ThemeSettingsType = "Oqtane.Themes.OqtaneTheme.ThemeSettings, Oqtane.Client",
           ContainerSettingsType = "Oqtane.Themes.OqtaneTheme.ContainerSettings, Oqtane.Client",
```

These types are related to razor files in the Theme:

Oqtane.Client\Themes\OqtaneTheme\Containers\ContainerSettings.razor
Oqtane.Client\Themes\OqtaneTheme\Themes\ThemeSettings.razor

At runtime the framework uses this information to inject the setting components.

**Reply by markdav-is** [reply]
I see how this is implimented now. I was looking for settings attached to theme for a site.  I'm looking in the wrong place. I'm rethinking if I even need those settings at all.

**Comment by thabaum** [comment]
@sbwalker Would it make sense to add a place to add Theme Settings per site as well as per page?  In the section I produced a screenshot of earlier for site wide settings of a theme?

**Reply by sbwalker** [reply]
@thabaum the Setting Scope option in the Oqtane Theme contains "Site" or "Page" as options. So it is already possible to have theme settings per site.
